certbot-nginx/certbot_nginx/parser.py

```diff
@@ -241,6 +241,24 @@ def filedump(self, ext='tmp', lazy=True):
             except IOError:
                 logger.error("Could not open file for writing: %s", filename)
 
+    def has_ssl_on_directive(self, vhost):
+        """Does vhost have ssl on for all ports?
+
+        :param :class:`~certbot_nginx.obj.VirtualHost` vhost: The vhost in question
+
+        :returns: True if 'ssl on' directive is included
+        :rtype: bool
+
+        """
+        server = vhost.raw
+        for directive in server:
+            if not directive:
```

Comment:
Great I like it! Then we should probably also fix this in parse_server, in parser.py (in a different PR).